tests/unit/bokeh/models/test_annotations.py

```diff
@@ -241,6 +242,20 @@ def test_BoxAnnotation() -> None:
         "top_units",
     ], LINE, FILL, HATCH)
 
+def test_BoxAnnotation_accepts_datetime() -> None:
+    obj = BoxAnnotation(left = datetime(2018,8,7,0,0),
+                        right = datetime(2018,8,7,0,0),
+                        top = datetime(2018,8,7,0,0),
+                        bottom = datetime(2018,8,7,0,0),
+                       )
```

Comment:
I don't want get too distracted with stylistic discussions, especially given that other tests advertise the above style, but to me this is almost the hardest one to work with and maintain. This would be much better:
```suggestion
    obj = BoxAnnotation(
        left = datetime(2018, 8, 7, 0, 0),
        right = datetime(2018, 8, 7, 0, 0),
        top = datetime(2018, 8, 7, 0, 0),
        bottom = datetime(2018, 8, 7, 0, 0),
    )
```